### PR TITLE
Use mark tag for highlights

### PR DESCRIPTION
**Pull request by RoyEJohnson**
For: https://github.com/openstax/unified/issues/2647
For: https://openstax.atlassian.net/browse/DISCO-111
For: https://github.com/openstax/unified/issues/2718
For: https://github.com/openstax/unified/issues/2792
For: https://openstax.atlassian.net/browse/DISCO-120
For: https://openstax.atlassian.net/browse/DISCO-152
**Note 25-Mar**
The PR is now using a Highlighter release tag and can be merged whenever.

### CODE REVIEW

**Review by jivey — CHANGES_REQUESTED** [review]
The es and pl translation placeholders need to be updated from english, did you want to place an automated translation in there for now and tag other people for review?

**Comment by RoyEJohnson** [comment]
Yeah, the code changes aren't ready for review yet. I just wanted Malar to
look at them to see if it was working as expected.

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "@craco/craco": "<7",
     "@formatjs/intl-pluralrules": "^5.2.3",
     "@openstax/event-capture-client": "^2.0.2",
-    "@openstax/highlighter": "1.13.0",
+    "@openstax/highlighter": "https://github.com/openstax/highlighter#use-mark-dist-branch",
```
[the remark this answers is not in the document]
Comment (RoyEJohnson) [reply package.json:12]: When it's been tested, we should merge the Highlighter branch and create a new named release tag, then plug that in here.

**Review by jivey — APPROVED** [review]

**Comment by RoyEJohnson** [comment]
@jivey The last commit addresses a separate but closely issue.

File: src/app/messages/en/messages.json [hunk 2]
```diff
@@ -63,8 +63,10 @@
   "i18n:highlighter:edit-note:label": "Edit highlighted note",
   "i18n:highlighter:highlight:end": "End of {style} highlight",
   "i18n:highlighter:highlight:start": "Start {style} highlight",
+  "i18n:highlighter:highlight:start-selected": "Start selected {style} highlight",
```
Comment (jivey) [inline src/app/messages/en/messages.json:66]: This doesn't seem to match the other messages?

**Review by jivey — CHANGES_REQUESTED** [review]

File: src/app/messages/en/messages.json (hunk 2 above)
Comment (RoyEJohnson) [reply src/app/messages/en/messages.json:66]: @jivey I'm not clear on what other messages should be matched and how. This is a message to indicate the start of the selected highlight (as opposed to other, not-currently-selected (orange) ones).

Comment (jivey) [reply src/app/messages/en/messages.json:66]: Oh I was just looking at the other `i18n:highlighter:highlight:start-selected` keyed messages in the diff and was not paying attention to what the style transformation was doing, sorry about that.

**Review by jivey — APPROVED** [review]
Looks good, not sure if we need translation sign offs.

**Comment by RoyEJohnson** [comment]
@paulinaszczucinska could you verify or correct the translations for the "-selected" messages? These are distinguishing the currently-selected highlight from the other, not-currently-selected highlights. The parts in braces handle output of a color name; it should be ok as long as it's in the right place in the message.
```
  "i18n:highlighter:highlight:start": "Początek {style, select, blue {niebieskiego} green {zielonego} pink {różowego} purple {fioletowego} yellow {żółtego} other {{style}}} zakreślenia",
  "i18n:highlighter:highlight:start-selected": "Początek {style, select, blue {niebieskiego} green {zielonego} pink {różowego} purple {fioletowego} yellow {żółtego} other {{style}}} wybranego",
  "i18n:highlighter:highlight:end:search": "Koniec zakreślenia wyniku wyszukiwania",
  "i18n:highlighter:highlight:start:search": "Początek zakreślenia wyniku wyszukiwania",
  "i18n:highlighter:highlight:start-selected:search": "Początek wybranego wyniku wyszukiwania",
```

**Comment by paulinaszczucinska** [comment]
@RoyEJohnson translations look good, thank you!

**Comment by RoyEJohnson** [comment]
@jivey I've made substantial updates here. Highlighter is updated to add the language as a class to a marked highlight so that `::before` and `::after` can be used by the screenreader, rather than real (span) elements.
I did not see a straightforward way to use `intl` to generate the content, so I brought the screenreader translations into the PageContent code.

**Review by jivey — CHANGES_REQUESTED** [review]
Would placing the translation in a data attribute and using `attr` on `content` work for this? https://developer.mozilla.org/en-US/docs/Web/CSS/content#attrx
If so, it would be a nice way to keep the strings out of CSS.

**Comment by RoyEJohnson** [comment]
Excellent suggestion, @jivey ! It made a lot of things simpler.

**Review by jivey — APPROVED** [review]
Looks good!

**Pull request merged**